keras/backend/tensorflow_backend.py

```diff
@@ -193,6 +201,72 @@ def set_session(session):
     _SESSION = session
 
 
+# DEVICE MANIPULATION AND PROBING
+
+class _TfDeviceCaptureOp(object):
+    """Class for capturing the TF device scope."""
+
+    def __init__(self):
+        self.device = None
+
+    def _set_device(self, device):
+        """This method captures TF's explicit device scope setting."""
+        self.device = device
+
+
+def _get_current_tf_device():
+    """Return explicit device of current context, otherwise returns `None`.
+
+    # Returns
+        If the current device scope is explicitly set, it returns a string with 
+        the device (`CPU` or `GPU`). If the scope is not explicitly set, it will
+        return `None`.
+    """
+    g = tf.get_default_graph()
+    op = _TfDeviceCaptureOp()
+    g._apply_device_functions(op)
+    return op.device
+
+
+def is_current_explicit_device(device_type):
+    """Check if the current device is explicitly set on the device type specified.
+
+    # Args
+        device_type (str): `GPU` or `CPU` (case-insensitive).
+
+    # Returns
+        bool: if the current device scope is explicitly set on the device type.
+    """
+    device_type = device_type.upper()
+    if device_type not in ['CPU', 'GPU']:
+        raise ValueError('device_type should be either "CPU" or "GPU".')
+    device = _get_current_tf_device()
+    return (device is not None and device.device_type == device_type.upper())
+
+
+def get_available_gpus():
```

Comment:
Please make this method private (unless there is a rationale for making it part of the public API).